TypeCobol/Compiler/CodeElements/CodeElement.cs

```diff
@@ -473,6 +474,32 @@ public int Channel
             }
         }
 
+        /// <summary>
+        /// Get the line number in the main source for this CodeElement.
+        /// If it belongs to an imported COPY, the line is the one declaring the COPY directive in the main source
+        /// </summary>
+        /// <returns>The line number</returns>
+        public int GetLineInMainSource()
+        {
+            if (ConsumedTokens.Count > 0)
+            {
+                var firstToken = ConsumedTokens[0];
+                if (firstToken is ImportedToken importedToken)
+                {
+                    if (importedToken.CopyDirective.TextNameSymbol.TokensLine is CodeElementsLine codeElementsLine)
```

Comment:
Why the type checks here and at line 494 ? `LineIndex` is defined by `ITextLine`.

Reply:
Type checks not needed => removed